src/Rector/AbstractRector.php

```diff
@@ -208,8 +209,13 @@ final public function enterNode(Node $node)
             $this->changedNodeScopeRefresher->reIndexNodeAttributes($node);
         }
 
+        $startTime = microtime(true);
+        $previousMemory = memory_get_peak_usage(true);
```

Comment:
this should only defined on debug mode, I prefer `$this->rectorOutputStyle->isDebug()` is saved to variable to ensure no repetitive calls, start with above `printDebugCurrentFileAndRule()` call...